crates/nu-parser/src/parse.rs

```diff
@@ -517,13 +517,18 @@ fn parse_interpolated_string(
     (call, error)
 }
 
+// TODO this comment is copied from parse_arg and wrong here
 /// Parses the given argument using the shape as a guide for how to correctly parse the argument
 fn parse_external_arg(
     registry: &dyn SignatureRegistry,
     lite_arg: &Spanned<String>,
 ) -> (SpannedExpression, Option<ParseError>) {
     if lite_arg.item.starts_with('$') {
-        return parse_full_column_path(&lite_arg, registry);
+        if !lite_arg.item.contains('.'){
+            return (SpannedExpression::new(Expression::variable(lite_arg.item.clone(), lite_arg.span.clone()), lite_arg.span), None);
+        }else{
+            return parse_full_column_path(&lite_arg, registry);
```

Comment:
This function can parse invocations. For example, `$(echo $it | str collect)`, which doesn't include a `.`.

I think if you want this behaviour, it would be best to somehow make it work within `parse_full_column_path`, but I also think that this is working as intended. Basically, an `Expression::variable` isn't meant to stand alone, but rather as part of an `Expression::path`.

@jonathandturner is our parser person, and is best equipped to comment on this.

Reply:
Thanks for clarifying.
I thought something like `echo $b` would result in Expressions of type: Expression::Command and Expression::Variable.

If your AST is not built like that, its totaly fine. I was expecting things differently (as sketched out in the code quoted above).